frontend/android/fastlane/Fastfile

```diff
@@ -13,26 +13,176 @@
+
+KEYSTORE_PATH = ENV['KEYSTORE_PATH']
+KEYSTORE_PASSWORD = ENV['KEYSTORE_PASSWORD']
+
+KEYSTORE_KEY_ALIAS_BAYERN = ENV['KEYSTORE_KEY_ALIAS_BAYERN']
+KEYSTORE_KEY_ALIAS_NUERNBERG = ENV['KEYSTORE_KEY_ALIAS_NUERNBERG']
+KEYSTORE_KEY_PASSWORD_BAYERN = ENV['KEYSTORE_KEY_PASSWORD_BAYERN']
+KEYSTORE_KEY_PASSWORD_NUERNBERG = ENV['KEYSTORE_KEY_PASSWORD_NUERNBERG']
+
+
 default_platform(:android)
 
 platform :android do
-  desc "Runs all the tests"
-  lane :test do
-    gradle(task: "test")
+  desc "Download and decrypt the JKS"
+  lane :keystore do
+    ensure_env_vars(
+      env_vars: ['CREDENTIALS_GIT_REPOSITORY_URL', 'CREDENTIALS_KEYSTORE_PATH', 'CREDENTIALS_DIRECTORY_PATH', 'CREDENTIALS_KEYSTORE_PASSWORD']
+    )
+
+    puts("Cloning repository with keystore")
+
+    unless File.exists? File.expand_path(CREDENTIALS_DIRECTORY_PATH)
+      sh("git clone #{CREDENTIALS_GIT_REPOSITORY_URL} #{CREDENTIALS_DIRECTORY_PATH}")
+    end
+
+    puts("Decrypting keystore")
+
+    sh("gpg --verbose --passphrase #{CREDENTIALS_KEYSTORE_PASSWORD} --pinentry-mode loopback \\
+                -o ~/#{KEYSTORE_PATH} -d #{CREDENTIALS_KEYSTORE_PATH}")
   end
 
-  desc "Submit a new Beta Build to Crashlytics Beta"
-  lane :beta do
-    gradle(task: "clean assembleRelease")
-    crashlytics
-  
-    # sh "your_script.sh"
-    # You can also use other beta testing services here
+  desc "Validate Play Store Key"
+  lane :validate_play_store_key do
+    validate_play_store_json_key(
+      json_key_data: ENV["GOOGLE_SERVICE_ACCOUNT_JSON"]
+    )
   end
 
-  desc "Deploy a new version to the Google Play"
-  lane :deploy do
-    gradle(task: "clean assembleRelease")
-    upload_to_play_store
+  desc "Build Android App"
+  lane :build do |options|
+    version_code = options[:version_code]
+    version_name = options[:version_name]
+    flavor = options[:flavor]
+    build_config_name = options[:build_config_name]
+
+    if [version_name, version_code].include?(nil)
+      raise "'nil' passed as parameter! Aborting..."
+    end
+
+    gradle_system_properties = {
+      # 2GB Gradle + 1GB dex + 2-2.5GB RN < 6GB of circleci resource class medium+
+      :"org.gradle.jvmargs" => "-Xms512m -Xmx2024m",
+      :"org.gradle.daemon" => false
+    }
+
+    if ENV['TOTAL_CPUS']
+      # Gradle uses the wrong cpu count from the host (e.g. 36)
+      gradle_system_properties["org.gradle.workers.max"] = ENV['TOTAL_CPUS']
+    end
+
+    sh("fvm flutter pub run build_runner build --define df_build_config=name=#{build_config_name}")
+
+    gradle(
+      task: "bundle",
+      flavor: flavor,
+      flags: " -Ptarget=lib/main.dart",
+      build_type: "Release",
+      properties: {
+        :VERSION_CODE => version_code,
+        :VERSION_NAME => version_name,
+        :BUILD_CONFIG_NAME => build_config_name,
+        :KEYSTORE_PATH => KEYSTORE_PATH,
+        :KEYSTORE_KEY_ALIAS => flavor == 'Bayern' ? KEYSTORE_KEY_ALIAS_BAYERN : KEYSTORE_KEY_ALIAS_NUERNBERG,
```

Comment:
i think a more generic implementation would be nice, if we gonna add a third project/flavor . Same for `KEYSTORE_KEY_PASSWORD`

Reply:
For more flavours i highly recommend to reuse the Bayern-upload key and NOT create a new one for every app.

Reply:
ok so if we use the `KEYSTORE_KEY_ALIAS_BAYERN` for all except of Nuernberg we should probably switch the condition
`:KEYSTORE_KEY_ALIAS => flavor == 'Nuernberg' ? KEYSTORE_KEY_ALIAS_NUERNBERG : KEYSTORE_KEY_ALIAS_BAYERN,`